lib/ansible/modules/remote_management/manageiq/manageiq_user.py

```diff
@@ -0,0 +1,237 @@
+        """
+        user = self.manageiq.find_collection_resource_by('users', userid=userid)
+        if not user:  # user doesn't exist
+            return dict(
+                changed=False,
+                msg="User {userid} does not exist in manageiq".format(userid=userid))
+        try:
+            url = '{api_url}/users/{user_id}'.format(api_url=self.manageiq.api_url, user_id=user['id'])
+            result = self.manageiq.client.post(url, action='delete')
+            return dict(changed=True, msg=result['message'])
+        except Exception as e:
+            self.manageiq.module.fail_json(msg="Failed to delete user {userid}: {error}".format(userid=userid, error=e))
+
+    def user_update_required(self, user, username, group_id, email):
+        """ Returns True if the username, group id or email passed for the user
+            differ from the user's existing ones, False otherwise.
+        """
+        if username is not None and user['name'] != username:
+            return True
+        if group_id is not None and user['current_group_id'] != group_id:
+            return True
+        if email is not None and user.get('email') != email:
+            return True
+        return False
+
+    def update_user_if_required(self, user, userid, username, group_id, password, email):
+        """Updates the user in manageiq.
+
+        Returns:
+            the created user id, name, created_on timestamp,
+            updated_on timestamp, userid and current_group_id
+        """
+        if not self.user_update_required(user, username, group_id, email):
+            return dict(
+                changed=False,
+                msg="User {userid} already exist, no need for updates".format(userid=userid))
+        url = '{api_url}/users/{user_id}'.format(api_url=self.manageiq.api_url, user_id=user.id)
+        resource = {'userid': userid, 'group': {'id': group_id}}
+        if username is not None:
+            resource['name'] = username
+        if password is not None:
+            resource['password'] = password
+        if email is not None:
+            resource['email'] = email
+
+        try:
+            result = self.manageiq.client.post(url, action='edit', resource=resource)
+        except Exception as e:
+            self.manageiq.module.fail_json(msg="Failed to update user {userid}: {error}".format(userid=userid, error=e))
+        return dict(
+            changed=True,
+            msg="Successfully updated the user {userid}: {user_details}".format(userid=userid, user_details=result))
+
+    def create_user(self, userid, username, group_id, password, email):
+        """Creates the user in manageiq.
+
+        Returns:
+            the created user id, name, created_on timestamp,
+            updated_on timestamp, userid and current_group_id
+        """
+        url = '{api_url}/users'.format(api_url=self.manageiq.api_url)
+        resource = {'userid': userid, 'name': username, 'password': password,
```

Comment:
When `password` is not specified (this is a requirement for MIQ external auth) what get's passed to the MIQ API? Seem like it should test `password is not None` before building the json.